src/test/ui/fuckit/no-borrowck-dead-code.rs

```diff
@@ -0,0 +1,9 @@
+// compile-flags: -Z fuckit -Ztreat-err-as-bug=1
+// run-pass
+fn dead_code(s: &str) -> &'static str {
+    s
+}
+
+fn main() {
+    println!("he he he")
+}
```

Comment:
If we actually call `dead_code`, this will be UB but "work" as long as it is called with a static string, right? The "fully built version" of this would ideally have a way of catching misuses at runtime. It'd be useful to also emit the compile errors even though they are ignored, maybe marking them as such (but that can be worked around by calling rustc twice 🤷).

Reply:
No, borrowck will still run when creating the MIR for the function. This is pretty much unavoidable as one of the outputs of borrowck is the list of upvars for closures, which is necessary for building the MIR for closures and their callers.